send.c

```diff
@@ -648,6 +650,20 @@ static void add_ra_option_abro(struct safe_buffer *sb, struct AdvAbro const *abr
 	safe_buffer_append(sb, &abro, sizeof(abro));
 }
 
+static void add_ra_option_capport(struct safe_buffer *sb, const char *captive_portal)
+{
+	/* +2 for the ND_OPT_CAPTIVE_PORTAL and the length (each occupy one byte) */
+	size_t const capport_strlen = strlen(captive_portal);
+	size_t const capport_bytes = capport_strlen + 2;
+	size_t const capport_len = (capport_bytes + 7) / 8;
```

Comment:
I don't understand the _divide by eight_.
I think it is some memory alignment thing.
But, I would have expect _modulo eight_, so with a `%` as operator.

Reply:
It kinda is. The logic was copied from for example the existing add_ra_option_sllao() function. (The same is present in add_ra_something implementations).

capport_len is not the number of bytes, but as specified in the RA protocol the length byte contains the number of bytes divided by 8. Partial blocks of 8 are to be padded using zero bytes to the next full multiple of 8 - this "(bytes + 7) / 8" implements this "increase by one for partially filled blocks of 8".

The padding itself then happens a bit below with the safe_buffer_pad() call.

Reply:
Thanks for elaborating.
Thanks for educating me  `:-)`